cogs/starboard.py

```diff
@@ -0,0 +1,128 @@
+from typing import List, Optional
+import discord
+from discord.ext import commands
+from discord import ui
+
+STARBOARD_CHANNEL_ID = 1133260871049691257  # 1139734225390669875
+ON_PHONE_EMOJI_STR = "<:OnPhone:1062142401973588039>"  # "<:OnP:1139749947357544500>"
+
+
+class Starboard(commands.Cog):
+    def __init__(self, bot: discord.Client):
+        self.starboard_emoji_str = ON_PHONE_EMOJI_STR
+        self.threshold = 1
+        # Starboarded Message ID -> ID of the message the bot sent.
+        self.starboard_msgs = dict()
+        self.starboard_channel = bot.get_channel(STARBOARD_CHANNEL_ID)
+
+    async def cog_load(self):
+        await super().cog_load()
+        print("Starboard Cog loaded.")
+
+    def _get_title(self, react: discord.Reaction) -> str:
+        return f"{self.starboard_emoji_str} x **{react.count}** |{react.message.channel.mention}"
+
+    async def _get_open_msg_view(self, msg: discord.Message) -> ui.View:
+        btn = ui.Button(label="Jump", url=msg.jump_url)
+        v = ui.View().add_item(btn)
+
+        if msg.type != discord.MessageType.reply:
+            return v
+
+        reply = msg.reference.cached_message or await msg.channel.fetch_message(
+            msg.reference.message_id
+        )
+        return v.add_item(ui.Button(label="Context", url=reply.jump_url))
```

Comment:
personally Im not a fan of having this button, since you already include the message being replied to in the starboard. the Jump button will take the user to the starboarded message and if it was a reply it's easy to see the context then